### PR TITLE
[Bugfix] Loss funcs for multitask learning should also be added as torch nn.module

### PR DESCRIPTION
**Pull request by classicsong**
*Issue #, if available:*
The loss funcs of GSgnnMultiTaskSharedEncoderModel are not added as nn.modules of GSgnnMultiTaskSharedEncoderModel. This will cause device error in some cases.

*Description of changes:*
Adding loss funcs as a nn.ModuleDict() of GSgnnMultiTaskSharedEncoderModel.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Review by jalencato — APPROVED** [review]
LGTM
